### PR TITLE
Allow the user to load the plugin module

### PR DESCRIPTION
**Pull request by splatteredbits**

### CODE REVIEW

File: Rivet/Functions/Import-RivetPlugin.ps1 [hunk 1]
```diff
@@ -5,32 +5,54 @@ function Import-RivetPlugin
...
+
+    $commands = & {
+        foreach( $moduleName in $moduleNames )
+        {
+            Write-Timing -Message "  Get Commands  BEGIN  $($moduleName)"
+            if( -not (Get-Module -Name $moduleName) )
+            {
+                $msg = ("Unable to load plugins from module ""$($moduleName)"": the module is not loaded. Please " +
+                        'call "Import-Module" to load this module before running Rivet. If you want Rivet to load the ' +
+                        'module for you, use the ""PluginPaths"" setting and set it to a list of paths to modules ' +
```
Comment (pnstickne) [inline Rivet/Functions/Import-RivetPlugin.ps1]: `.."Pluginpaths"..`

File: Rivet/Functions/Import-RivetPlugin.ps1 [hunk 2]
```diff
@@ -5,32 +5,54 @@ function Import-RivetPlugin
...
 
-            Write-Timing -Message ('  Import {0}  BEGIN' -f $pluginPath)
-            $module = Import-Module -Name $pluginPath -Global -Force -PassThru
-            Write-Timing -Message ('  Import {0}  END' -f $pluginPath)
+            Write-Timing -Message "  Import  BEGIN  $($pluginPath)"
+            Import-Module -Name $pluginPath -Global -Force -PassThru |
+                Select-Object -ExpandProperty 'Name' |
+                Write-Output
+            Write-Timing -Message "  Import  END    $($pluginPath)"
+
```
Comment (pnstickne) [inline Rivet/Functions/Import-RivetPlugin.ps1]: Newline.

File: Rivet/Functions/Import-RivetPlugin.ps1 [hunk 3]
```diff
@@ -5,32 +5,54 @@ function Import-RivetPlugin
...
+        }
+
+        $ModuleName | Write-Output
```
Comment (pnstickne) [inline Rivet/Functions/Import-RivetPlugin.ps1:36]: Debugging or expected all time?

**Review by pnstickne — DISMISSED** [review]

File: Rivet/Functions/Get-RivetConfig.ps1 [hunk 4]
```diff
@@ -87,48 +80,39 @@ function Get-RivetConfig
...
             # Set the configuration value as an integer.
-            $AsInt,
+            [switch]$AsInt,
 
-            [Parameter(Mandatory,ParameterSetName='AsList')]
-            [Switch]
+            [Parameter(Mandatory,ParameterSetName='AsArray')]
             # Set the configuration value as a list of strings.
-            $AsList,
+            [switch]$AsArray,
```
Comment (pnstickne) [inline Rivet/Functions/Get-RivetConfig.ps1:96]: Usage changes?

File: Rivet/Functions/Get-RivetConfig.ps1 [hunk 5]
```diff
@@ -328,22 +313,30 @@ function Get-RivetConfig
...
             else
             {
-                New-Object 'Rivet.Configuration.Database' $dbName,$_.FullName
+                [Rivet.Configuration.Database]::New($dbName, $databaseInfo.FullName) | Write-Output
             }
-        } | 
-        ForEach-Object { $configuration.Databases.Add( $_ ) }
+        }
+
+        foreach( $rivetDatabase in $rivetDatabases )
```
Comment (pnstickne) [inline Rivet/Functions/Get-RivetConfig.ps1]: No 'AddRange'-style?

File: Rivet/Functions/Import-RivetPlugin.ps1 (hunk 3 above)
Comment (splatteredbits) [reply Rivet/Functions/Import-RivetPlugin.ps1:36]: It's just making it explicit that the module name is being returned from the current block to the outer block.

```
$moduleNames = & {
    # snip
    $ModuleName | Write-Output
}
```

File: Rivet/Functions/Get-RivetConfig.ps1 (hunk 4 above)
Comment (splatteredbits) [reply Rivet/Functions/Get-RivetConfig.ps1:96]: No. I thought `AsArray` was clearer what was being returned than `AsList`. I didn't want people to think a `Collections.Generic.List` was getting returned.

**Review by pnstickne — APPROVED** [review]

**Pull request merged**